lib/breathe/client.rb

```diff
@@ -1,43 +1,45 @@
 module Breathe
   class Client
+    attr_reader :api_key, :last_response
+
     BASE_URL = "https://api.breathehr.com/v1/"
 
     def initialize(api_key:)
       @api_key = api_key
     end
 
     def absences
-      @_absences ||= Absences.new(self)
-    end
-
-    def get(url, url_opts = {})
-      request(:get, url, url_opts)
+      @absences ||= Absences.new(self)
     end
 
-    private
+    def response(type, path, args)
+      response = request(type, path, query: args)
 
-    attr_reader :api_key, :environment
+      return Response.new(response, path) if /^2+/.match?(response.status.to_s)
```

Comment:
Could we query the response for a successful status code in a clearer way that might encourage a unit test or 2? Perhaps after we initialise our response object?

```
breathe_response = request(…)
response = Response.new(response: breathe_response, path: path)
return response if response.success?
```